### PR TITLE
Added pre-commit hook to ensure API keys aren't committed

### PR DESCRIPTION
**Pull request by NachoSoto**
Fixes [CSDK-273].

[CSDK-273]: https://revenuecats.atlassian.net/browse/CSDK-273?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: scripts/pre-commit.sh [hunk 1]
```diff
@@ -29,6 +29,26 @@ run_swiftlint() {
   fi
 }
 
+verify_no_included_apikeys() {
+  SCRIPT_DIR=$(cd $(dirname $(readlink -f "${BASH_SOURCE[0]}")) && pwd)
```
Comment (NachoSoto) [inline scripts/pre-commit.sh:33]: `readlink -f` follows the link from `.git/hooks/pre-commit` back to `scripts/pre-commit.sh` if executed through there. If not it returns the same file.

File: scripts/pre-commit.sh [hunk 2]
```diff
@@ -29,6 +29,26 @@ run_swiftlint() {
...
+  FILES_TO_CHECK=(
+    "${SCRIPT_DIR}/../Tests/BackendIntegrationTests/Constants.swift"
+    "${SCRIPT_DIR}/../Examples/MagicWeather/MagicWeather/Constants.swift"
+    "${SCRIPT_DIR}/../Examples/MagicWeatherSwiftUI/Shared/Constants.swift"
+  )
+  PATTERN="\"REVENUECAT_API_KEY\""
```
Comment (NachoSoto) [inline scripts/pre-commit.sh:39]: This simply looks for this pattern.

File: scripts/pre-commit.sh [hunk 3]
```diff
@@ -29,6 +29,26 @@ run_swiftlint() {
...
+
+  for i in "${FILES_TO_CHECK[@]}" 
+  do
+    grep -q $PATTERN $i
+    FOUND=$?
+    if [ $FOUND -ne 0 ]; then
+      echo "Leftover API Key found in '$(basename $i)'. Please remove."
+      exit $FOUND
```
Comment (NachoSoto) [inline scripts/pre-commit.sh:47]: And this is the output.

File: scripts/pre-commit.sh (hunk 1 above)
Comment (aboedo) [reply scripts/pre-commit.sh:33]: maybe we can add that as a comment?

**Review by aboedo — APPROVED** [review]
👏

File: scripts/pre-commit.sh (hunk 1 above)
Comment (NachoSoto) [reply scripts/pre-commit.sh:33]: Done.

**Pull request merged**